### DISCUSSION TITLE
Is it possible to set environment variables in a file?

### DISCUSSION THREAD

**Discussion by nmills3** [itzg/docker-minecraft-server#2492] [Q&A] [answered]
Is it possible to put the environment variables in some kind of file instead of using docker compose or a giant docker run command?

**Comment by itzg** [comment, marked as answer]
Yes it is. With a compose project, a `.env` file is automatically included if present or can be explicitly referenced:

https://docs.docker.com/compose/environment-variables/set-environment-variables/#substitute-with-an-env-file

With `docker run` you can reference env files with `--env-file`:

https://docs.docker.com/engine/reference/commandline/run/#env

**Reply by nmills3** [reply]
thanks
